src/providers/search/search.ts

```diff
@@ -11,16 +12,61 @@ export class SearchProvider {
     public urlProvider : UrlProvider,
   ) {}
 
-  driverNumberSearch(driverNumber: string): Observable<any> {
+  driverNumberSearch(driverNumber: string, isLDTM: boolean = false): Observable<any> {
     return this.http.get(
-      this.urlProvider.getTestResultServiceUrl().concat(`?driverNumber=${driverNumber}`),
+      this.urlProvider.getTestResultServiceUrl()
+        .concat(`?driverNumber=${driverNumber}&isLDTM=${isLDTM}`),
     );
   }
 
-  applicationReferenceSearch(applicationReference: String): Observable<any> {
+  applicationReferenceSearch(applicationReference: string, isLDTM : boolean = false): Observable<any> {
     return this.http.get(
-      this.urlProvider.getTestResultServiceUrl().concat(`?applicationReference=${applicationReference}`),
+      this.urlProvider.getTestResultServiceUrl()
+        .concat(`?applicationReference=${applicationReference}&isLDTM=${isLDTM}`),
     );
   }
 
+  advancedSearch(advancedSearchParams: AdvancedSearchParams): Observable<any> {
+
+    const httpParams: string [] = [];
+
+    advancedSearchParams.isLDTM ?
+     httpParams.push('isLDTM=true') : httpParams.push('isLDTM=false');
+
+    if (advancedSearchParams.startDate) {
+      httpParams.push(`startDate=${advancedSearchParams.startDate}`);
+    }
+
+    if (advancedSearchParams.endDate) {
+      httpParams.push(`endDate=${advancedSearchParams.endDate}`);
+    }
+
+    if (advancedSearchParams.staffNumber) {
+      httpParams.push(`staffNumber=${advancedSearchParams.staffNumber}`);
+    }
+
+    if (advancedSearchParams.costCode) {
+      httpParams.push(`dtcCode=${advancedSearchParams.costCode}`);
+    }
+
+    return this.http.get(
+      this.urlProvider.getTestResultServiceUrl()
+        .concat(this.buildQueryString(httpParams)),
```

Comment:
Are there any plans for sanitising those search inputs params at any point? Both at frontend and at the backend? 
Because it smells SQL injection 👃

Reply:
Yeah we have a sub task to sanatise everything :)